src/table/runtime.rs

```diff
@@ -374,6 +434,44 @@ pub const GLOBAL_VARIABLE: Guid = Guid::from_values(
     [0x00, 0xe0, 0x98, 0x03, 0x2b, 0x8c],
 );
 
+/// Unique key for a variable.
+#[cfg(feature = "exts")]
+pub struct VariableKey {
+    name: Vec<u16>,
+    /// Unique identifier for the vendor.
+    pub vendor: Guid,
+}
+
+#[cfg(feature = "exts")]
+impl VariableKey {
+    /// Name of the variable.
+    pub fn name(&self) -> core::result::Result<&CStr16, FromSliceWithNulError> {
+        CStr16::from_u16_with_nul(&self.name)
+    }
+}
+
+#[cfg(feature = "exts")]
+impl fmt::Debug for VariableKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "VariableKey {{ name: ")?;
+
+        match self.name() {
+            Ok(name) => write!(f, "\"{}\"", name)?,
+            Err(err) => write!(f, "Err({:?})", err)?,
+        }
+
+        write!(f, ", vendor: ")?;
+
+        if self.vendor == GLOBAL_VARIABLE {
+            write!(f, "GLOBAL_VARIABLE")?;
+        } else {
+            write!(f, "{}", self.vendor)?;
+        }
+
+        write!(f, " }}")
```

Comment:
Have you considered using something like [`DebugStruct`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html) for this `Debug` impl? Or is it not flexible enough?

Reply:
Oh that's neat, I didn't know about DebugStruct.

Reply:
It looks like in order to use that I'd have to make temporary string allocations, and it would insert some unwanted quotes, so I've left the code unchanged.

I have made one adjustment though, which is to change this to a Display impl and just derive the Debug impl. This formatter is trying to make the info pretty and human readable, so I think Display is probably a better fit.

For reference, here's the Debug vs Display output:
```
VariableKey { name: [83, 101, 116, 117, 112, 77, 111, 100, 101, 0], vendor: Guid { a: 2347032417, b: 37834, c: 4562, d: [170, 13, 0, 224, 152, 3, 43, 140] } }
VariableKey { name: "SetupMode", vendor: GLOBAL_VARIABLE }
```

Reply:
Yep, the `Debug` vs `Display` change matches the distinction in the rest of the library. I think it's fine to keep the implementation as it is, without `DebugStruct`.